walletnode/services/artworkregister/task.go

```diff
@@ -716,38 +733,83 @@ func (task *Task) connectToTopRankNodes(ctx context.Context) error {
 	task.UpdateStatus(StatusConnected)
 	task.nodes = nodes
 
+	// Send all meshed supernode info to nodes - that will be used to node send info to other nodes
+	meshedSNInfo := []types.MeshedSuperNode{}
+	for _, node := range nodes {
+		meshedSNInfo = append(meshedSNInfo, types.MeshedSuperNode{
+			NodeID: node.PastelID(),
+			SessID: node.SessID(),
+		})
+	}
+
+	for _, node := range nodes {
+		err = node.MeshNodes(ctx, meshedSNInfo)
+		if err != nil {
+			nodes.DisconnectAll()
+			return errors.Errorf("could not send info of meshed nodes: %w", err)
+		}
+	}
 	return nil
 }
 
+func (task *Task) sendRegMetadata(ctx context.Context) error {
+	if task.creatorBlockHash == "" {
+		return errors.New("empty current block hash")
+	}
+
+	if task.Request.ArtistPastelID == "" {
+		return errors.New("empty creator pastelID")
+	}
+
+	regMetadata := &types.NftRegMetadata{
+		BlockHash:       task.creatorBlockHash,
+		CreatorPastelID: task.Request.ArtistPastelID,
+	}
+
+	return task.nodes.SendRegMetadata(ctx, regMetadata)
+}
+
 func (task *Task) probeImage(ctx context.Context) error {
 	log.WithContext(ctx).WithField("filename", task.Request.Image.Name()).Debug("probe image")
 
 	// Send image to supernodes for probing.
 	if err := task.nodes.ProbeImage(ctx, task.Request.Image); err != nil {
 		return errors.Errorf("send image: %w", err)
 	}
+	signatures := [][]byte{}
+	// Match signatures received from supernodes.
+	for i := 0; i < len(task.nodes); i++ {
+		verified, err := task.pastelClient.Verify(ctx, task.nodes[i].FingerprintAndScoresBytes, string(task.nodes[i].Signature), task.nodes[i].PastelID(), pastel.SignAlgorithmED448)
+		if err != nil {
+			return errors.Errorf("probeImage: pastelClient.Verify %w", err)
+		}
+
+		if !verified {
+			task.UpdateStatus(StatusErrorSignaturesNotMatch)
+			return errors.Errorf("node[%s] signature doesn't match", task.nodes[i].PastelID())
+		}
+
+		signatures = append(signatures, task.nodes[i].Signature)
+	}
+	task.signatures = signatures
 
 	// Match fingerprints received from supernodes.
 	if err := task.nodes.MatchFingerprintAndScores(); err != nil {
 		task.UpdateStatus(StatusErrorFingerprintsNotMatch)
 		return errors.Errorf("fingerprints aren't matched :%w", err)
 	}
+
 	task.fingerprintAndScores = task.nodes.FingerAndScores()
 	task.UpdateStatus(StatusImageProbed)
 
 	// As we are going to store the the compressed figerprint to kamedila
-	// so we calculated the hash base on the compressed fingerprint print also
-	fingerprintsHash, err := sha3256hash(task.fingerprintAndScores.ZstdCompressedFingerprint)
+	// so we calculated the hash based on the compressed fingerprints as well
+	fingerprintsHash, err := utils.Sha3256hash(task.fingerprintAndScores.ZstdCompressedFingerprint)
```

Comment:
`task.fingerprintAndScores = task.nodes.FingerAndScores()`
The fingerprintAndScores.ZstdCompressedFingerprint was not set in task.nodes.FingerAndScores() - should set it.

Reply:
@nguyen202 not sure what you mean, sry

Reply:
I did not see any where to set `ZstdCompressedFingerprint`, so we have null value here.

Reply:
@nguyen202 pls push the fix for this?